seatunnel-core/seatunnel-spark-starter/seatunnel-spark-2-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/TransformExecuteProcessor.java

```diff
@@ -125,6 +130,22 @@ private Dataset<Row> sparkTransform(SeaTunnelTransform transform, Dataset<Row> s
         transform.setTypeInfo(seaTunnelDataType);
         StructType structType =
                 (StructType) TypeConverterUtils.convert(transform.getProducedType());
+        StructField[] newStructFields =
```

Comment:
Hi, can you move this logic to TypeConverterUtils.convert?

Reply:
Modifications in TypeConverterUtils.convert may affect other modules.
The root cause of this problem is that for `TIMESTAMP` Type, the `TimestampType` in spark is returned in `TypeConverterUtils.convert`, while the `Long` Type is returned in `InternalRowConverter.convert`.
Inconsistent types cause spark to report an error.